Movie-App-Group-7/Movie-App-Group-7/UI/Detail Movie/DetailMovieViewController.swift

```diff
@@ -0,0 +1,124 @@
+    }
+    
+    private func retrieveDetailMovie(_ movieId: String) {
+        
+        viewModel.retrieveMovie(movieId: movieId) { [weak self] result in
+            switch result {
+            case .success(let optionalMovie):
+                guard let movie = optionalMovie else {
+                    self?.navigationController?.popViewController(animated: true)
+                    return
+                }
+                self?.bindView(movie)
+            case .failure(let message):
+                print(message)
+            }
+        }
+    }
+    
+    private func configureScrollView() {
+        scrollView.contentInsetAdjustmentBehavior = .never
+        scrollView.contentInset = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)
+    }
+    
+    private func bindView(_ movie: Movie) {
+        guard let url = URL(string: movie.movieBanner) else { return }
+        guard let urlPoster = URL(string: movie.image) else { return }
+    
+        loadImageUsingKingfisher(url: url, urlPoster: urlPoster)
+        movieTitleLabel.text = movie.title
+        movieJapaneseTitleLabel.text = movie.originalTitle + "(\(movie.originalTitleRomanised))"
+        movieDurationLabel.text = movie.formattedDuration
+        movieYearLabel.text = movie.releaseDate
+        movieDirectorLabel.text = "By \(movie.director)"
+        movieDescriptionLabel.text = movie.description
+    }
+    
+    private func loadImageUsingKingfisher(url: URL, urlPoster: URL) {
+        isDownloadingImage(true, movieImageContainerView)
+        isDownloadingImage(true, movieBannerContainerView)
+        
+        movieImageView.kf.setImage(
+            with: url,
+            placeholder: nil,
+            options: nil,
+            progressBlock: { [weak self] _, _ in
+                self?.isDownloadingImage(true, self?.movieImageContainerView)
+            },
+            completionHandler:  { [weak self] result in
+                switch result {
+                case .success(_):
+                    self?.isDownloadingImage(false, self?.movieImageContainerView)
+                case .failure(let error):
+                    self?.isDownloadingImage(false, self?.movieImageContainerView)
+                    print(error)
+                }
+            }
+        )
+        
+        movieBannerImageView.kf.setImage(
+            with: urlPoster,
+            placeholder: nil,
+            options: nil,
+            progressBlock: { [weak self] _, _ in
+                self?.isDownloadingImage(true, self?.movieBannerContainerView)
+            },
+            completionHandler:  { [weak self] result in
+                switch result {
+                case .success(_):
+                    self?.isDownloadingImage(false, self?.movieBannerContainerView)
+                case .failure(let error):
+                    self?.isDownloadingImage(false, self?.movieBannerContainerView)
+                    print(error)
+                }
+            }
+        )
```

Comment:
ini sebenernya bisa lu rubah jadi satu fungsi tinggal dirubah parameter imageViewnya aja.
```suggestion
       private func loadImageUsingKingfisher(url: URL, urlPoster: URL) {
        downloadImage(for : movieImageContainerView, with : url)
        downloadImage(for : movieBannerImageView, with : urlPoster)
        }
        private func downloadImage(for imageView: UIImageView, with url: URL) {
                isDownloadingImage(true, imageView)
       imageView.kf.setImage(
            with: url,
            placeholder: nil,
            options: nil,
            progressBlock: { [weak self] _, _ in
                self?.isDownloadingImage(true, imageView)
            },
            completionHandler:  { [weak self] result in
                switch result {
                case .success(_):
                    self?.isDownloadingImage(false, imageView)
                case .failure(let error):
                    self?.isDownloadingImage(false, imageView)
                    print(error)
                }
            }
        )
        }
```